src/v/cluster/topic_table.cc

```diff
@@ -678,6 +694,11 @@ topic_table::wait_for_changes(ss::abort_source& as) {
     if (!_pending_deltas.empty()) {
         ret_t ret;
         ret.swap(_pending_deltas);
```

Comment:
Does anyone think this is better than just `ret_t ret(std::move(_pending_deltas));`? 

I guess the swap variant is more explicit in that it doesn't require you to understand than a moved-from vector is guaranteed to be empty.

Reply:
> I guess the swap variant is more explicit in that it doesn't require you to understand than a moved-from vector is guaranteed to be empty.

yeh, guaranteeing that the moved-from vector is the important bit. the example i use is that if `_pending_deltas` happened to be a const-ref then the compiler will not complain and `std::move(const ref)` will really be a copy.

here i'd avoid swap, and be even more explicit: `auto ret = std::exchange(_pending_deltas, {})`

Reply:
Really good point, @dotnwat - I hadn't considered that the move might sometimes fail and result in a copy: normally that's "only" a performance problem but where we rely on the well-defined moved-from state as we do here it would also be a hard-to-spot bug.